boards/shared/Test/Src/Test_ErrorTable.cpp

```diff
@@ -0,0 +1,539 @@
+        ASSERT_EQ(
+            EXIT_CODE_OK,
+            App_SharedErrorTable_SetError(
+                error_table, DEFAULT_FSM_NON_CRITICAL_ERROR, true));
+        ASSERT_EQ(
+            EXIT_CODE_OK,
+            App_SharedErrorTable_SetError(
+                error_table, DEFAULT_PDM_NON_CRITICAL_ERROR, true));
+    }
+
+    std::vector<enum ErrorBoard> GetAllBoards(void)
+    {
+        return std::vector<enum ErrorBoard>{
+            BMS, DCM, DIM, FSM, PDM,
+        };
+    }
+
+    // The error choices here are arbitrary. We just need to pick something for
+    // the helper functions.
+    const enum ErrorId    DEFAULT_CRITICAL_ERROR           = BMS_CRIT;
+    const enum ErrorId    DEFAULT_NON_CRITICAL_ERROR       = BMS_NON_CRIT;
+    const enum ErrorBoard DEFAULT_CRITICAL_ERROR_BOARD     = BMS;
+    const enum ErrorBoard DEFAULT_NON_CRITICAL_ERROR_BOARD = BMS;
+
+    const enum ErrorId DEFAULT_BMS_CRITICAL_ERROR = BMS_CRIT;
+    const enum ErrorId DEFAULT_DCM_CRITICAL_ERROR = DCM_CRIT;
+    const enum ErrorId DEFAULT_DIM_CRITICAL_ERROR = DIM_CRIT;
+    const enum ErrorId DEFAULT_FSM_CRITICAL_ERROR = FSM_CRIT;
+    const enum ErrorId DEFAULT_PDM_CRITICAL_ERROR = PDM_CRIT;
+
+    const enum ErrorId DEFAULT_BMS_NON_CRITICAL_ERROR = BMS_NON_CRIT;
+    const enum ErrorId DEFAULT_DCM_NON_CRITICAL_ERROR = DCM_NON_CRIT;
+    const enum ErrorId DEFAULT_DIM_NON_CRITICAL_ERROR = DIM_NON_CRIT;
+    const enum ErrorId DEFAULT_FSM_NON_CRITICAL_ERROR = FSM_NON_CRIT;
+    const enum ErrorId DEFAULT_PDM_NON_CRITICAL_ERROR = PDM_NON_CRIT;
+
+    struct ErrorTable *error_table;
+    struct ErrorList   error_list;
+};
+
+TEST_F(SharedErrorTableTest, is_board_in_list)
+{
+    struct ErrorBoardList board_list;
+
+    // The board exists in the list
+    board_list.boards[0]  = BMS;
+    board_list.boards[1]  = DCM;
+    board_list.boards[2]  = DIM;
+    board_list.boards[3]  = FSM;
+    board_list.boards[4]  = PDM;
+    board_list.num_boards = 5;
+    ASSERT_TRUE(App_SharedError_IsBoardInList(&board_list, BMS));
+
+    // The board does not exist in the list
+    board_list.boards[0]  = BMS;
+    board_list.boards[1]  = BMS;
+    board_list.boards[2]  = BMS;
+    board_list.boards[3]  = BMS;
+    board_list.boards[4]  = BMS;
+    board_list.num_boards = 5;
+    ASSERT_FALSE(App_SharedError_IsBoardInList(&board_list, DCM));
+
+    // The board we are looking for has multiple entries in the list
+    board_list.boards[0] = BMS;
+    board_list.boards[1] = BMS;
+    board_list.boards[2] = DIM;
+    board_list.boards[3] = FSM;
+    board_list.boards[4] = PDM;
+    ASSERT_TRUE(App_SharedError_IsBoardInList(&board_list, BMS));
+
+    // The board exists in the list but is out-of-bound
+    board_list.boards[0]  = BMS;
+    board_list.boards[1]  = DCM;
+    board_list.boards[2]  = DIM;
+    board_list.boards[3]  = FSM;
+    board_list.boards[4]  = PDM;
+    board_list.num_boards = 4;
+    ASSERT_FALSE(App_SharedError_IsBoardInList(&board_list, PDM));
+}
```

Comment:
bit of a nit; but each case here should really be a separate test case. There are several other similar cases below where there are multiple clearly distinct things being tested in a single test case (they're even already effectively named with a comment!).

Reply:
Yeah I kinda felt similarly when I was writing these, I'll spend some time to break things down